### ISSUE TITLE
[Question]  Is there a live document online Containing all the list of rules available in the catalog  along with their respective descriptions

### ISSUE THREAD

**Issue by allurisriakhil** [forcedotcom/sfdx-scanner#976]
Hi Team,

We want to know if there is any Page online that consists of all the list of rules available in the code analyzer catalog along with their respective descriptions?

I am aware of the commands that are provided in the documentation to check the list of rules and to describe each rule but this becomes difficult for us track the changes manually whenever a new version is released.

Or please let me know if there is any command that can provide the list of rules along with the description instead of finding description of each rule individually.

Thanks

**Comment by johnbelosf** [comment]
hi @allurisriakhil have you tried running [sfdx scanner:rule:list](https://forcedotcom.github.io/sfdx-scanner/en/v3.x/scanner-commands/list/)?

**Comment by johnbelosf** [comment]
Also as Code Analyzer uses multiple engines, the engines' respective docs will usually contain a list of rules.

For instance for [PMD](https://pmd.github.io/latest/pmd_rules_apex.html)
For Graph Engine you can see in our [docs](https://forcedotcom.github.io/sfdx-scanner/en/v3.x/salesforce-graph-engine/rules/)

**Comment by johnbelosf** [comment]
Certainly. We do have plans for this in our backlog, but nothing planned for the immediate future, so wanted to make sure you're aware of what we have available now. I'll mark this as a feature request. Thank you!

**Comment by stephen-carter-at-sf** [comment]
@allurisriakhil

Circling back on this, with code-analyzer v5 we now have
```
sf code-analyzer rules -r all -f allRules.json
```
that you can use to create a json file that describes all the rules.

Does this satisfy your request?
